.github/workflows/push-to-stage.yml

```diff
@@ -0,0 +1,44 @@
+name: Push to Staging
+
+on: 
+  push:
+  pull_request:
+    types:
+      - opened
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Extract branch name
+      shell: bash
+      run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
+      id: extract_branch
+    # Github uses different branch var names for push vs PR event types
+    - name: Extract branch name (PR Only)
+      if: github.event.action == 'opened'
+      shell: bash
+      run: echo "##[set-output name=branch;]$(echo ${GITHUB_HEAD_REF#refs/heads/})"
+      id: extract_branch_pr
+    - uses: actions/checkout@v2
+    - name: Install dependencies
+      run: npm install
+    - uses: ./
+    - name: Run tests
+      run: npm run test
+    - uses: jakejarvis/s3-sync-action@master
+      if: github.event_name == 'push'
+      with:
+        args: --acl public-read --follow-symlinks
+      env:
+        AWS_S3_BUCKET: ${{ secrets.AWS_S3_BUCKET }}
+        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        SOURCE_DIR: 'generated' 
+        DEST_DIR: 'trackerblocking/config/staging/${{ steps.extract_branch.outputs.branch }}'
```

Comment:
Is the output of the `extract_branch_pr` step ever used?

Reply:
Yes. There's a step to add a comment to the PR with the hosted config location. That's where that output is used. It's done on the PR opened event so it only happens once.